### PR TITLE
Fix semgrep findings

### PR DESCRIPTION
**Pull request by Sakchai-Refinitiv**
## Description

Fix Semgrep findings to keep EF maintaining code hygiene and to ensure apps adopting EF security compliant.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

JIRA: [ELF-2232](https://jira.refinitiv.com/browse/ELF-2232)

### CODE REVIEW

File: .github/workflows/semgrep.yml [hunk 1]
```diff
@@ -27,7 +27,13 @@ jobs:
...
           echo "VERBOSE_OPTION=$VERBOSE" >> $GITHUB_ENV
       - name: Scan
-        run: semgrep ci ${{ env.VERBOSE_OPTION }}
+        run: |
+          semgrep ci ${{ env.VERBOSE_OPTION }}
+          BBlue='\033[1;34m'
+          Color_Reset='\033[0m'
+          echo ""; echo -e "📝 ${BBlue}View results ${{ github.repository }} in Semgrep Cloud Platform${Color_Reset}"
+          echo "  https://semgrep.dev/orgs/-/findings?repo=Refinitiv%2Frefinitiv-ui"
+          echo "  https://semgrep.dev/orgs/-/supply-chain?q=refinitiv-ui"
```
Comment (wattachai-lseg) [inline .github/workflows/semgrep.yml]: this link add filtering to our main branches only.
```suggestion
          echo "https://semgrep.dev/orgs/-/supply-chain?q=Refinitiv%2Frefinitiv-ui&ref=refs%2Fheads%2Fv7%2Cv6"
```

**Comment by bualoy-napat** [comment]
LGTM!

**Review by bualoy-napat — DISMISSED** [review]

File: .github/workflows/semgrep.yml (hunk 1 above)
Comment (Sakchai-Refinitiv) [reply .github/workflows/semgrep.yml]: According to our discussion, we update the links by filtering the repository only.

**Comment by wattachai-lseg** [comment]
pending ReDos issue using security lib?

**Review by bualoy-napat — DISMISSED** [review]

File: .github/workflows/semgrep.yml [hunk 2]
```diff
@@ -27,7 +27,13 @@ jobs:
...
           echo "VERBOSE_OPTION=$VERBOSE" >> $GITHUB_ENV
       - name: Scan
-        run: semgrep ci ${{ env.VERBOSE_OPTION }}
+        run: |
+          semgrep ci ${{ env.VERBOSE_OPTION }}
+          BBlue='\033[1;34m'
+          Color_Reset='\033[0m'
+          echo ""; echo -e "📝 ${BBlue}View results ${{ github.repository }} in Semgrep Cloud Platform${Color_Reset}"
+          echo "  https://semgrep.dev/orgs/-/findings?repo=Refinitiv%2Frefinitiv-ui"
+          echo "  https://semgrep.dev/orgs/-/supply-chain?q=Refinitiv%2Frefinitiv-ui"
```
Comment (wattachai-lseg) [inline .github/workflows/semgrep.yml]: let's remove supply chain link as there is no official support.

**Comment by wattachai-lseg** [comment]
could you update `.semgrepconfig` by adding space between keys and values in the `tags` list?
```yml
tags:
    - asset: 250242
    - prod_reporting_branch: v6
    - prod_reporting_branch: v7
```
[This config is a YML](https://github.com/returntocorp/semgrep/blob/develop/CHANGELOG.md#added-45) and the file's seemingly not working due to the missing space between keys and values.

**Comment by Sakchai-Refinitiv** [comment]
> could you update `.semgrepconfig` by adding space between keys and values in the `tags` list?
> 
> ```yaml
> tags:
>     - asset: 250242
>     - prod_reporting_branch: v6
>     - prod_reporting_branch: v7
> ```
> 
> [This config is a YML](https://github.com/returntocorp/semgrep/blob/develop/CHANGELOG.md#added-45) and the file's seemingly not working due to the missing space between keys and values.

Done

**Review by wattachai-lseg — DISMISSED** [review]

**Review by wattachai-lseg — APPROVED** [review]
LGTM. Just need to verify result on semgrep dashboard.

**Review by wattachai-lseg — APPROVED** [review]

**Pull request merged**